File: src/main/java/pl/alpheratzteam/obfuscator/api/transformer/Transformer.java
package pl.alpheratzteam.obfuscator.api.transformer;

import org.objectweb.asm.Opcodes;
import org.objectweb.asm.tree.ClassNode;

import java.util.Map;

/**
 * @author Unix on 31.08.2019.
 */
public interface Transformer extends Opcodes {

    void transform(Map<String, ClassNode> classMap);

    String getName();

}

File: src/main/kotlin/pl/alpheratzteam/obfuscator/api/transformer/Transformer.kt
package pl.alpheratzteam.obfuscator.api.transformer

import pl.alpheratzteam.obfuscator.Obfuscator

/**
 * @author Unix
 * @since 16.12.2020
 */

interface Transformer {

    /**
     * This method transform some classes by obfuscator.
     * @param obfuscator as instance of obfuscator.
     * @see [ClassNode] class to transform.
     */
    fun transform(obfuscator: Obfuscator)

}